### PR TITLE
Fixed incorrect bounds of snapshot when UIAppearance properties cause...

### PR DESCRIPTION
**Pull request by jgongo**
… the view to resize. Fix for #6

### CODE REVIEW

File: FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDemoTests.m [hunk 1]
```diff
@@ -63,6 +63,28 @@ - (void)testViewSnapshotWithUIAppearance
   FBSnapshotVerifyView(control, nil);
 }
 
+- (void)testViewSnapshotWithUIAppearanceResizing
+{
+  // If this works properly you should be able to read the whole "Click me!" title in the snapshot
```
Comment (freak4pc) [inline FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDemoTests.m]: No need for the comment here IMO - it's obvious from the snapshots themselves

Comment (jgongo) [reply FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDemoTests.m]: You are right, removed :)

**Comment by jgongo** [comment]
The build failed with the following error:

    Test target FBSnapshotTestCaseDemoTests encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted)

This usually means an incorrectly configured job, or the simulator not launching fast enough. Can you please check it? Thanks!!

**Comment by alanzeino** [comment]
Try rebasing against the latest HEAD commit?

I'll patch this and run it on our test suite to see if anything breaks.

**Comment by jgongo** [comment]
I already rebased it against the latest HEAD commit

**Comment by jgongo** [comment]
@alanzeino anything I can do on my side to speed up things?

**Comment by alanzeino** [comment]
@jgongo as for the change, I have to run it on a serious test suite (not the small one in the demo app here) to be sure it's good for existing tests. Haven't had the time just yet.

As for the simulator problem, haven't looked at it. Might have to create and boot a specific simulator for this, but that's a bit of an effort. Other PRs seem just fine, so I don't know why this one triggered that problem. I don't know how travis ci works in terms of virtualized or static hardware but I would expect that each environment is clean and requested simulators are 'new' from the perspective of the job. Anyway, not something I'll likely look at soon if other CI builds pass. I recommend just retriggering.

**Comment by jgongo** [comment]
@alanzeino I've experienced this kind of errors with Travis seemingly under a big load. It seems the tests are tried to be run before the simulator is ready and they fail because of that. I don't think it has anything to do with my modifications.

How should I retrigger the check? With projects that I own I just go to Travis and restart the build, but I can't do it in this case.

**Comment by alanzeino** [comment]
I've restarted the build for you. Let's see how it goes.

**Comment by jgongo** [comment]
It worked 😄 🎆 🎉

**Comment by freak4pc** [comment]
Despite the fact it works, still need to fulfill this, I think :) 

> @jgongo as for the change, I have to run it on a serious test suite (not the small one in the demo app here) to be sure it's good for existing tests. Haven't had the time just yet.

This is a good change, but older tests might have been counting on this "wrong" behavior and it would break them. It's best to try on some large base of screens. I have about 400 snapshot tests in our code base, will hopefully be able to help later if @alanzeino can't.

**Comment by jgongo** [comment]
I'll have to wait a bit more, I guess :) Anyway I understand your concerns. Let me know if I can do anything else to help. Thanks a lot!

**Comment by alanzeino** [comment]
Ran on 29755 tests, no failures.

**Comment by freak4pc** [comment]
@alanzeino Kinda out-of-context but how are you running so many tests? We're running about 600 of them and it's a significant portion of our CI time. Did you guys build any tricks around this?

**Comment by jgongo** [comment]
@freak4pc I just have a few snapshots tests, but I find them to be pretty quick. In my case I test individual components, so I don't even have to run the application to run them. I even have the UI components in a separate pod (as I expect to reuse some of them in Apple TV) and I have the graphical unit tests there. Are testing this way or do you test whole screens in your real application?

**Comment by alanzeino** [comment]
@freak4pc those tests are split up over a 100+ test targets, most of which are library test bundles (not application test bundles) which allows us to run them concurrently with buck+xctool without a simulator on as many threads as the host machine has.

**Comment by freak4pc** [comment]
Very interesting. I would really love to learn more about this but I'm guessing this isn't the right context for this. If you could point me to any interesting read about this, that would be great, if you have a quick minute :) The main interesting question is about using library test bundles vs. application ones, and how UIView-based components are tested in that case.

**Pull request merged**